### ISSUE TITLE
Provider fails to handle account level grant if an Application is installed and has grants

### ISSUE THREAD

**Issue by SarahU** [Snowflake-Labs/terraform-provider-snowflake#1962]
**Provider Version**

0.67.0

**Terraform Version**

1.4.6

**Describe the bug**

We have existing snowflake_account_grant resources.

We then installed app: https://app.snowflake.com/marketplace/listing/GZT1Z7J3SP/ntt-data-cost-optimizer-for-snowflake,
and followed the instructions to grant it permissions.

The next plan we ran, generated the following error: 

```shellsession
╷
│ Error: unknown grantee type APPLICATION
│ 
│   with module.snowflake_base.snowflake_account_grant.create_user,
│   on snowflake_base/account_grants.tf line 1, in resource "snowflake_account_grant" "create_user":
│    1: resource "snowflake_account_grant" "create_user" {
│ 
╵
```

**Expected behavior**

No error - Terraform was working fine before the application was installed. Uninstalling the application, and hence removing its grants, fixes the issues (but then we don't have the app anymore..).

Expected behaviour is that the provider does not error out, for grant types it doesn't understand.

**Code samples and commands**

Below is an example of an existing grant in our app, but any account grant triggers this issue

```terraform

resource "snowflake_account_grant" "create_user" {
  provider  = snowflake.security_admin
  privilege = "CREATE USER"

  roles = [
    snowflake_role.my_provisioner.name,
  ]

  with_grant_option      = false
  enable_multiple_grants = false
}

```

Then we installed and ran the applications commands:

```sql
GRANT IMPORTED PRIVILEGES ON DATABASE SNOWFLAKE TO APPLICATION <APP_NAME>;

GRANT EXECUTE MANAGED TASK ON ACCOUNT TO APPLICATION <APP_NAME>;

GRANT EXECUTE TASK ON ACCOUNT TO APPLICATION <APP_NAME>;
```

This creates  grants in Snowflake, granted_to = Application
<img width="914" alt="Screenshot 2023-07-19 at 21 14 46" src="https://github.com/Snowflake-Labs/terraform-provider-snowflake/assets/245346/33b84222-0195-47ba-8598-57fd1e39e836">

Triggering a plan on our exisitng Snowflake now generates the error.

**Additional context**

Reviewing the provider code (latest) it seems to not be aware of the Application grantee type. It would be good for the provider to simply ignore types it does not understand.
See code: https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/pkg/resources/grant_helpers.go#L161-L193

Looks similar to this issue: https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1863

**Comment by sfc-gh-swinkler** [comment]
@SarahU have you tried using the [grant_privileges_to_role](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/grant_privileges_to_role) resource instead? The `snowflake_account_grant` resource has been deprecated in favor of this new resource. That should fix your problem. Example drop-in replacement for your use case:

```
resource "snowflake_grant_privileges_to_role" "g1" {
  privileges = ["CREATE USER"]
  role_name  = snowflake_role.my_provisioner.name
  on_account = true
  with_grant_option = false
}
```

Note that we are also working on creating a new `snowflake_grant_privileges_to_application`, which would grant permissions to applications.

**Comment by maxwiesner** [comment]
Seeing it with warehouse grants as well:
<img width="763" alt="Screenshot 2023-08-28 at 8 30 52 AM" src="https://github.com/Snowflake-Labs/terraform-provider-snowflake/assets/70331375/78c8b3b1-bcce-4fba-af23-6a3a119a5ed0">

**Comment by sfc-gh-swinkler** [comment]
This is fixed as of 0.70.1

**Comment by sfc-gh-jcieslak** [comment]
Hey 👋
Closing the issue as it's referring to the deprecated resource and we're not supporting them (and it seems like it was already fixed, but not confirmed). Please, try to use the latest provider version with the non-deprecated resources. If the issue persists on the non-deprecated resource, please create a new GitHub issue. The [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/docs/technical-documentation/resource_migration.md) may help during migration to the latest resources.
